peer/peer.go

```diff
@@ -433,6 +433,11 @@ type Peer struct {
 	cfg     Config
 	inbound bool
 
+	// These fields track the peer's whitelisted status and ban score.
+	isWhitelisted bool
```

Comment:
This doesn't make sense to me to be on the peer.  It is a function of the ban manager and is something that should be housed by it.